bitseq/sequence.go

```diff
@@ -240,8 +251,11 @@ func (h *Handle) set(ordinal uint32, any bool, release bool) (uint32, error) {
 			bytePos, bitPos = ordinalToPos(ordinal)
 		} else {
 			if any {
-				bytePos, bitPos, err = getFirstAvailable(h.head)
+				bytePos, bitPos, err = getFirstAvailable(h.head, start)
 				ret = posToOrdinal(bytePos, bitPos)
+				if end < ret {
```

Comment:
It is not going to fail.
The range limits are evaluated/used only when the flag `any` is true. That's why I set them only during the `SetAny()` and `SetAnyInRange()` API calls. `h.set()` is the inner function which handles the four public API, it ends up having those many inputs. But being an internal function I did not see a problem or a refactor need, as long as it works.